crates/fee/src/lib.rs

```diff
@@ -267,6 +267,159 @@ pub mod pallet {
             Self::withdraw_from_reward_pool::<T::VaultRewards, T::VaultStaking>(&vault_id, &nominator_id, index)?;
             Ok(().into())
         }
+
+        /// Changes the issue fee percentage (only executable by the Root account)
+        ///
+        /// # Arguments
+        ///
+        /// * `origin` - signing account
+        /// * `fee` - the new fee
+        #[pallet::weight(<T as Config>::WeightInfo::set_issue_fee())]
+        #[transactional]
+        pub fn set_issue_fee(
+            origin: OriginFor<T>,
+            fee: UnsignedFixedPoint<T>,
+        ) -> DispatchResultWithPostInfo {
+            ensure_root(origin)?;
+            Self::_set_issue_fee(fee);
```

Comment:
Imo there is no need to have this `_set_issue_fee` function - I would just inline it. The only reason to have it be a function in the pallet impl imo is when you want to be able to call the function from other pallets, which is not the case here. Having it as a separate function just adds clutter.

I'm guessing you probably took some of the vault registry setters as example, which uses the same construction that you did. I would actually be in favor of changing those as well (not in this PR), but it's slightly annoying in some of those cases since they are actively used in tests.